.github/workflows/unit_testing_wf.yaml

```
@@ -0,0 +1,23 @@
+name: macrophage-striation-unittests
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  unit-test-core:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: conda-incubator/setup-miniconda@v2
```

While it's optional, I recommend adding a step `name` when you can to help outline what each step is doing to increase readability (purposes are sometimes less evident based on the `uses` action reference). For example:
```suggestion
      - name: Setup conda env
        uses: conda-incubator/setup-miniconda@v2
```